### ISSUE TITLE
webPush.sendNotifications timesout while using inside aws lambda

### ISSUE THREAD

**Issue by aashutoshbane** [web-push-libs/web-push#638]
**Code implemented in AWS Lambda**:
```
this.options = {
	vapidDetails: {
		privateKey: process.env.VAPID_PRIVATE_KEY,
		publicKey: process.env.VAPID_PUBLIC_KEY,
		subject: 'mailto:ashutosh.bane@****.in',
	},
	TTL: 60,
	timeout: 20000,
};

const payload = "Test notification";

return webPush.sendNotification(deviceDetails, payload, this.options)
	.then(() => ({ code: 200, error: null }))
	.catch((e) => {
		logDebug(`sendNotifications Erorr: `, e);
		return { code: e.statusCode, error: e };
	});
```
*Operating System*: Linux
*Node Version*: 10.23.10
*web-push Version*: ^3.4.4

#### Am I missing something?

**Comment by RichPitul** [comment]
I am doing the same thing in Google Cloud Functions; the notification is received by my computer instantly but it times out on the server.
Runtime is Node.js 12 
web-push 3.4.4
`
 const pushSubscription = {
   "endpoint":"XXXXXX",
    "keys":{
      "p256dh":"XXXX",
      "auth":"XX"
    }
 };
 const payload = {
   title:"Notification Title",
   text:"Notification Text",
   time: new Date().getTime()
 };
 const options = {  
  vapidDetails: {
    subject: 'mailto:...',
    publicKey: "XXXXXXXXXXXXXXXXXX",
    privateKey: "XXXXX"
  },  
}
const webpush = require('web-push');
exports.init = (req, res) => {  
  const jsonString = JSON.stringify(payload);  
  webpush.sendNotification(
    pushSubscription,
    jsonString,
    options
  ).then(() => ({ code: 200, error: null }));  
};
`

**Comment by TiddlyWin** [comment]
The implementation that I was working on suffered the same timeout fate, so I began checking to see if the Lambda could actually talk to the outside world. I used `https://github.com/mihaerzen/lambda-internet-access-checker`  this repo and replaced my lambda code for this test. It turned out that I hadn't configured my VPC, Subnets, Routing and NAT correctly. I'm not well versed within AWS but I followed `https://stackoverflow.com/a/55267891` this answer and managed to get AWS configured correctly.
